scripts/build_firewood.sh

```diff
@@ -15,45 +15,66 @@ FIREWOOD_PATH=${FIREWOOD_PATH:-"./build/firewood"}
 FIREWOOD_VERSION=${FIREWOOD_VERSION:-"main"}
 FIREWOOD_IGNORE_CHECKOUT=${FIREWOOD_IGNORE_CHECKOUT:-"false"}
 
+check_prerequisites() {
+    returnval=0
+    for exe in "$@"; do
+        if ! command -v "$exe" >/dev/null 2>&1; then
+            echo "❌ $exe is not on your path"
+            returnval=1
+        fi
+    done
+
+    return $returnval
+}
+
 import_firewood() {
-    local original_dir=$(pwd)
     local firewood_remote="${1}"
     local firewood_path="${2}"
     local firewood_version="${3}"
     if [[ -d ${firewood_path} ]]; then
-        echo "Firewood repo already exists: ${firewood_path}"
+        echo "✅ Firewood repo directory already exists at ${firewood_path}"
+        if ! [[ -f "${firewood_path}/Cargo.toml" ]]; then
+            echo "❌ The directory doesn't contain a valid Cargo.toml"
```

Comment:
What's the reason for this check? Did you run into an issue with it or just a normal sanity check when working with a rust repo?

Reply:
Normal sanity check. If someone creates, say, an empty directory this script won't be very helpful diagnosing it. This is a little better.